java/src/main/java/org/rocksdb/StringAppendOperator.java

```diff
@@ -18,6 +20,19 @@ public StringAppendOperator(char delim) {
         super(newSharedStringAppendOperator(delim));
     }
 
+    public StringAppendOperator(byte[] delim) {
+        super(newSharedStringAppendTESTOperator(delim));
```

Comment:
It seems a bad idea that, depending on the constructor argument, a different implementation of the string append opertor is used. I wonder if instead it would be better to either i) have a different class in the java API that alwys goes against `StringAppendTESTOperator` or ii) replace `SharedStringAppendOperator` with `SharedStringAppendTESTOperator` in all places here. Is there any scenario where a user would prefer `SharedStringAppendOperator` over `SharedStringAppendTESTOperator`?

Reply:
I think for the time being this approach is okay.

* `StringAppendOperator` implements the Associative Merge Operator
* `StringAppendTESTOperator` implements the Non-Associative Merge Operator API